### ISSUE TITLE
Integrated EQ on each fx line

### ISSUE THREAD

**Issue by gi0e5b06** [LMMS/lmms#4004]
It's quite common on mixing consoles to have at least an EQ on each channel.
![mixer_20171123](https://user-images.githubusercontent.com/30031949/33169262-492cb95a-d044-11e7-9a07-b1d3c4fa8dda.png)
Some code https://github.com/gi0e5b06/lmms/commit/dba523c5b7637548e09fb1f4e20a13778e0c1b38

**Comment by musikBear** [comment]
but could it not just be an option in settings:
"Create all new Mixer-channels with default 10bandEQ"
Or an other EQ -Could be user selected as *Default-EQ*
I think that the cost i realestate for two dials is high

**Comment by Z3R0C** [comment]
why? the selection is the on/off button. 

that's generally a good idea!

**Comment by gi0e5b06** [comment]
Thanks for the feedback.
Yes, an option in the configuration to show/hide the dials (actually there are three) would be nice because it may not be useful for every one.It's mostly for live performances, even if it could be used when composing. This is why I picked up the DJ EQ effect to implement it. Please note that this effect is not part of the effect chain and it is applied after it. With the current version, you can achieve the same by adding an effect in each effect chain but that's not so practical but the dials are not visible.

Creating a channel with default effects could be interesting but it's a different issue. It can also be done by using a template. Maybe a template is even better since you can configure those effects.

**Comment by michaelgregorius** [comment]
I agree with @musikBear that we should not slap a default EQ on each channel. Mixer channels in DAWs are different from analogue mixing consoles in that they offer much more flexibility due to their ability to use different plugins as their inserts. This has the obvious advantage that the user can choose which EQ (s)he wants to use on a certain channel.

The implementation shown in the screenshot would have disadvantages in case a user does not want to use the default EQ on the channels. In that case:
* The user will always have to make sure that the default EQ is turned off.
* The default EQ will only eat screen real estate.

# Track templates to the rescue
I think a much more flexible solution would be to implement track templates like they can for example be found in REAPER. Roughly they could work as follows:
* You create a track template as follows:
   * Create an empty track.
   * Add all the effects and plugins that should go into the template to the new track.
   * Right click the channel and select "Save track template...". This opens a dialog where you can choose a name for the template (that must not exist yet). If closed with "OK" the template will be stored in a template directory.
* If you want to create a new channel you can do so by selecting the template to use from a menu. This menu is populated from the templates found in the template directory.
* For even more convenience there should be an option to set a default track template in the options. This default track template would then be applied to all tracks that are created by a simple "New channel" option. If no default template is set an empty channel is created.
* Optional: Provide the option to apply track templates to existing tracks, either by removing all existing plugins from that track or by adding the plugins from the track template before or after the existing plugins.

**Comment by gi0e5b06** [comment]
I guess the main point was not understood: it's for live perfs (or quick changes for playing/emulating a different sound system). The EQs are not part of the song but belong to the LMMS configuration.
I already agreed the EQs should not always be visible (and when they are hidden, they are turned off).
Nothing is frozen, I'm experimenting. But I find it quite cool & useful so far.

Concerning the track templates (you probably mean channel templates), I have no idea if it would be useful or not (but I'm not against it). In practice, you set your channel only once per song and it is a quick op. But OTOH it would be useful to have predefined chains, so wait & see the code.

**Comment by vlad1777d** [comment]
Hello.
Personally I use no eq in mixer tracks in most of time. I just use low pass filter + high pass filter + compressor + (maybe) some effect.
I'm not sure it's needed by default.

**Comment by michaelgregorius** [comment]
@gi0e5b06 Yes, in the context of LMMS they should indeed be called channel templates instead of track templates.

Concerning live performances I would assume that most people would rather prefer to be able to map arbitrary controls from LMMS to hardware controllers. It's much easier to precisely twist lots of knobs in a short time using real knobs instead of using a mouse.

**Comment by Umcaruje** [comment]
I think this is an absolute waste of mixer real estate, a 3 knob eq is very imprecise since it would have fixed values for the frequencies. We have a graphical EQ which can be added with a matter of few clicks that is incredibly more powerful and useful than a mixer one. So a big :-1: on this from me. Both from a UI and UX perspective.

**Comment by musikBear** [comment]
@Umcaruje how are you on https://github.com/LMMS/lmms/issues/4004#issuecomment-346678387
I like that, so does many others (well 4...

**Comment by Umcaruje** [comment]
@musikBear I love @michaelgregorius's idea but I feel like it should be a seperate issue, since it could get buried with time. 

I didn't adress it in my comment tho, my comment was directed to the original issue.

**Comment by musikBear** [comment]
> it should be a seperate issue, since it could get buried with time.
Yes! True 
@michaelgregorius -Maby a new separate is a good idea

**Comment by michaelgregorius** [comment]
Channel templates are now covered by #4023.
